app/Core/DateParser.php

```diff
@@ -204,13 +204,13 @@ private function getValidDate($value, $format)
 
         if ($date !== false) {
             $errors = DateTime::getLastErrors();
-            if ($errors['error_count'] === 0 && $errors['warning_count'] === 0) {
+            if ($errors == false) {
                 $timestamp = $date->getTimestamp();
                 return $timestamp > 0 ? $timestamp : 0;
             }
         }
 
-        return 0;
+        return false;
```

Comment:
should this change too? if so, it might be good to update the doc string

Reply:
I think so as it is a static value, docstring updated :+1:  .

Reply:
I'm not that familiar with this code base, but my inclination is to think that int return type is correct, and not boolean - the other exit location appears to return an integer unless I am misunderstanding the method :)

Reply:
Thanks for the atention! its enough with the conditional fix, opened new [PR](https://github.com/kanboard/kanboard/pull/5116) to address that.